bootloader/bootloader.go

```diff
@@ -327,3 +322,15 @@ func removeKernelAssetsFromBootDir(bootDir string, s snap.PlaceInfo) error {
 
 	return nil
 }
+
+// ForGadget returns a bootloader matching
+func ForGadget(gadgetDir, rootDir string, opts *Options) (Bootloader, error) {
```

Comment:
in a follow up we can use this in InstallBootConfig

Reply:
I'll add a TODO which we can address with the changes I have here https://github.com/bboozzoo/snapd/commit/9b684dfadba34afd7199891ba8bfe0d6ec56b1d6